### PR TITLE
chore: move product settings page to server components

### PR DESCRIPTION
**Pull request by pandeymangg**
## What does this PR do?

- Moves the product settings `apps/web/app/(app)/environments/[environmentId]/settings/product/page.tsx` page to server components using server components and server actions.
- The server actions use internal services for performing queries and mutations to the db
- Adds a loading state to the product settings page

Screenshot 2023-08-12 at 1 44 51 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

Go to the settings page for a product

## Checklist

- [x] Added a screen recording or screenshots to this PR
- [x] Filled out the "How to test" section in this PR
- [x] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/(app)/environments/[environmentId]/settings/product/DeleteProduct/DeleteProduct.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+type DeleteProductProps = {
+  environmentId: string;
+  product: TProduct;
+};
+
+export default async function DeleteProduct({ environmentId, product }: DeleteProductProps) {
+  const session = await getServerSession(authOptions);
+
+  const profile = session ? await getProfile(session.user.id) : null;
+  const team = await getTeamByEnvironmentId(environmentId);
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/settings/product/DeleteProduct/DeleteProduct.tsx:19]: We do a lot of complicated database operations "just" to determine the role of the current user within the team. We also already have something we can use for that in the session:
```json
{
  "@formbricks/web:go": {
    "user": {
      "name": "Matti",
      "email": "mail@matti.sh",
      "id": "cll6p6pdo000019t83t36a3te",
      "createdAt": "2023-08-11T14:42:55.356Z",
      "onboardingCompleted": true,
      "teams": [
        {
          "id": "cll6p6pdo000119t84cao4qch",
          "role": "owner",
          "plan": "free"
        }
      ]
    }
  }
}
```

**Review by mattinannt — CHANGES_REQUESTED** [review]
@pandeymangg Thanks a lot for this PR 🎉💪🙏 I left you a few comments for things I'd like to get changed before we can merge it into main 😊

File: apps/web/app/(app)/environments/[environmentId]/settings/product/DeleteProduct/DeleteProduct.tsx
[the remark this answers is not in the document]
Comment (pandeymangg) [reply apps/web/app/(app)/environments/[environmentId]/settings/product/DeleteProduct/DeleteProduct.tsx:1]: got it

File: apps/web/app/(app)/environments/[environmentId]/settings/product/DeleteProduct/DeleteProduct.tsx (hunk 1 above)
Comment (pandeymangg) [reply apps/web/app/(app)/environments/[environmentId]/settings/product/DeleteProduct/DeleteProduct.tsx:19]: this is really cool, wish I had inspected the session object more 😅

**Review by mattinannt — APPROVED** [review]
@pandeymangg Thanks a lot for this refactor to server components 😊 I merged the latest changes from main, fixed conflicts and will now merge it into main :-)

**Pull request merged**